pkg/api/dashboard.go

```diff
@@ -72,7 +72,23 @@ func (hs *HTTPServer) TrimDashboard(c *models.ReqContext) response.Response {
 }
 
 func (hs *HTTPServer) GetDashboard(c *models.ReqContext) response.Response {
-	uid := web.Params(c.Req)[":uid"]
+	params := web.Params(c.Req)
+	path, ok := params["*"] // path based endpoint
+	uid := params[":uid"]
+
+	// Check for path based UIDs (from file systems)
+	if ok || uid == "" {
+		dto, err := hs.StorageService.GetDashboard(c, path)
+		if err != nil {
+			return response.Error(500, "error getting path dashboard", err)
+		}
+		if dto != nil {
+			c.TimeRequest(metrics.MApiDashboardGet)
+			return response.JSON(200, dto)
+		}
+		uid = path // assume the path is actually UID
+	}
```

Comment:
Since this looks to ignore everything else in this HTTP handler (Guardian, setting meta flags, etc), might as well have this in a separate HTTP handler?

Reply:
Good point -- I moved it to the /storage endpoint -- the reason it was here (before) was so it would fall though to the uid flavor.  But I think better if we support that from the frontend.  (if it has a slash hit one endpoint or the other)